### PR TITLE
Fix image test [WIP]

### PR DESCRIPTION
**Pull request by damienmarchal**
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
@fredroy, @guparan  In this PR just wanted to have the exception message related with the Image_test.

Here it is: 
```
unknown file
C++ exception with description "[instance(0,0,0,0,0x0,non-shared)] CImg<unsigned char>::load(): Failed to recognize format of file '/Users/sofa/jenkins/workspace/mac_clang-3.4_options/pr/all/applications/plugins/image/image_test/scenes/lena.jpg'." thrown in the test body.
```

Any idea why ? Could it be the image CImg library compiled without jpg support or something like that ? 


EDIT: Ok I fixed this one...it is because ImageMagick is not install on some machine and thus CImg does not handle jpg. @guparan (I have installed ImageMagick on sofa-centos). 

EDIT2: I updated the mac. I remember that CImg need to install external tool on windows. But don't remember which one...maybe it was imagemagick also. @guparan would you have some time to dig into that ?

**Comment by damienmarchal** [comment]
I also would like to get rid of the failure in Utils:
https://ci.inria.fr/sofa-ci/job/centos_clang-3.4_options/1522/testReport/

Here is the associated msg_warning:
[WARNING] [Utils::widenString()] Invalid or incomplete multibyte or wide character

Could it be because this one is changing the locale and depending on the system things goes wrong with the french character used in the string ?

**Comment by fredroy** [comment]
@damienmarchal 
According to the CMakeLists.txt, image does not enable jpeg support in CImg (it needs to add "-Dcimg_use_jpeg" compiler flag)

**Comment by damienmarchal** [comment]
@fredroy thanks for the info...

Here are the corresponding lines on windows: 
Adding Plugin CImgPlugin
-- Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR) 
-- Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR) 

So we either need to install the imagemagick approach (I assume when CIMg cannot do something with a file it calls imagemagick) or to install the JPEG/TIFF libraries on windows/centos/macos. 

I also wonder if CMakeLists.txt use conditionnal for JPEG why the tests are done unconditionally ?

**Comment by fredroy** [comment]
Ah, you must be careful to not confuse image and CImgPlugin. Both of them are using CImg but they are using it separately.
The warning you are noticing comes from CImgPlugin (which warns if it does not find libjpeg and libtiff).
CImgPlugin_test tests if JPEG or TIFF have been enabled (if not, of course it does not test it)

As for plugin image, I think it was developed mainly by people using imagemagick on Linux or mac (which is called automatically if CImg does not recognize the format I think).
On Windows, it is not really natural to have imagemagick into the PATH...

Anyway, the solutions are:
 - change lena.jpg by lena.bmp (it should not need any external library)
 - add imagemagick to the Windows CI
 - add libjpeg/libtiff to the external libraries of Sofa (like libpng) and add libjpeg/libpng support to the image plugin.

File: SofaKernel/framework/sofa/helper/system/FileMonitor_macosx.cpp [hunk 1]
```diff
@@ -119,17 +119,17 @@ These steps are explained in more detail in the sections that follow.
...
-        printf("... numEvent %", numEvents );
+        printf("... numEvent %d\n", numEvents );
 
         fflush(stdout);
     }
 
-    FSEventStreamRef    m_eventStream;
-    bool                m_changed;
+    FSEventStreamRef    m_eventStream ;
+    bool                m_changed {false} ;
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/helper/system/FileMonitor_macosx.cpp:127]: But alors your were not initialized ?

**Pull request closed without merging**